docs/developer-docs/gas-cost.md

```diff
@@ -1,100 +1,230 @@
+
+At a high level, canister calls can be visualized using the following diagram:
+
+![Canister calls overview](./_attachments/transaction-overview.png)
+
+Each type of canister call has a different cycles cost associated with it, and the canister responsible for paying the cycles varies based on the type of call.
+
+### Canister creation
+
+An initial fee is charged when a canister is first created. Canisters are created by deploying a project onto an instance of the IC replica (locally or on the mainnet), or a canister can be created by another canister that is already running on a replica instance. Canisters created by another canister are referred to as 'child canisters'. The cycles charge for creating a canister is charged to the canister that is being created.
+
+### Messaging
+
+Messaging refers to calls that are made to a canister's methods. A canister's messaging costs depend on the type of message, the size of the message's request and response bytes, and the number of messages.
+
+Additional factors of cost include the number of nodes in the subnet the canister is deployed on, the total number of users interacting with the canister, and the number of daily active users.
+
+Messaging cost drivers include:
+
+- Message type:
+    - Query calls: Query data and perform a 'read-only' operation. (There is currently no cost for query calls).
+    - Update calls: Makes changes to the canister's state and must go through consensus.
+    - Inter-canister calls: Calls made between canisters.
+    - [Calls made by timers and heartbeats](/docs/current/developer-docs/backend/periodic-tasks): Periodic calls that are reoccurring.
+- Calls made daily per active user.
+- Size of each message's request bytes.
+- Instructions executed per message.
+- Calls per message.
+- Size of the message's request and response bytes per call.
+
+### Execution
+
+Execution refers to the tasks that the canister executes and the number of instructions executed per task and per call. These tasks use compute resources and rely on 'best-effort compute', meaning they do not pre-allocate compute resources unless otherwise specified, as compute resources are scarce and expensive.
 
-## Resource reservation mechanism
+For canisters that specifically request compute resources, the priority of allocating those resources is achieved through a scheduling strategy. Priority is determined based on the canister's `compute_allocation` setting and the amount of time the canister has been waiting to be scheduled. A non-zero `compute_allocation` will give a canister a higher priority but does not guarantee execution in a given round. However, you will have to pay a fee in every round for this higher priority.
+
+Execution cost drivers include:
```

Comment:
I think we can drop this and below because this is not part of the pricing model and useful only for high-level estimation (which is in another page).

Reply:
Removed this text

Reply:
Looks like the text is still here.

Reply:
This text was removed:

`For canisters that specifically request compute resources, the priority of allocating those resources is achieved through a scheduling strategy. Priority is determined based on the canister's `compute_allocation` setting and the amount of time the canister has been waiting to be scheduled. A non-zero `compute_allocation` will give a canister a higher priority but does not guarantee execution in a given round. However, you will have to pay a fee in every round for this higher priority.`

Is there another portion in this section that should be removed?

Reply:
Chatted via Slack and revised.